src/System.Slices/src/System/Span.cs

```diff
@@ -283,45 +288,25 @@ public bool ReferenceEquals(Span<T> other)
 
         public override int GetHashCode()
         {
-            // TODO: Write something better
-            uint hash = unchecked((uint)Length);
-            foreach (T el in this)
+            var hashCode = Length ^ Offset.GetHashCode();
+            if (Object != null)
             {
-                hash = (hash >> 7) | (hash << 25);
-                hash ^= unchecked((uint)el.GetHashCode());
+                hashCode ^= Object.GetHashCode();
             }
-
-            return unchecked((int)(hash));
+            return hashCode;
         }
 
+        /// <summary>
+        /// Checks to see if two spans point at the same memory.  Note that
+        /// this does *not* check to see if the *contents* are equal.
+        /// </summary>
         public bool Equals(Span<T> other)
         {
-            if (ReferenceEquals(other))
-            {
-                return true;
-            }
-
-            if (this.Length != other.Length)
-            {
-                return false;
-            }
-
-            var nonBoxingComparer = EqualityComparer<T>.Default;
-            for (int i = 0; i < Length; i++)
-            {
-                if (!nonBoxingComparer.Equals(
-                    this.GetItemWithoutBoundariesCheck(i), other.GetItemWithoutBoundariesCheck(i)))
-                {
-                    return false;
-                }
-            }
-
-            return true;
+            return ReferenceEquals(other);
```

Comment:
@krwq  I did some research: none of the existing .NET collections implements Equals as it's content equality check.  [ArraySegment](http://referencesource.microsoft.com/#mscorlib/system/arraysegment.cs,f7a911466af4caf1) [ImmutableArray](https://github.com/dotnet/corefx/blob/5f8f64437cf36f3baf0438b72a3987d3a527308f/src/System.Collections.Immutable/src/System/Collections/Immutable/ImmutableArray%601.cs#L1045)
Only the struct-based collections/wrappers do implement IEqutable interface at all. Mostly to avoid boxing when used in any scenartio that uses the default comparers (like List.Contains etc).

Enumerable offers quite popular (but very slow!) SequenceEquals extension method that allows to compare content of two IEnumerables. That is why I believe that it would be intuitive for users to call this method for content equality check. 

What do you think on such solution:
1. Keep the existing ReferenceEquals method. Add both == != operators that would call it.
2. Implement IEqutable interface in explicit way. There is no boxing, people don't use it by accident.
3. Keep the SequenceEquals extensions methods to get the best performance for IEqutable value types (most common scenario?).
4. Rename the proposed StructuralEquals to BlockEquals to get fast and both type and memory safe comparison for slices of different primitive types.